### PR TITLE
Add description and platform to crawler run metadata [sc-7034]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

Add more information about the crawler

### 🤓 What?

- Add description and platform to crawler run metadata
- update pre-commit hooks version, since the older hooks don't work with the latest pre-commit (on python 3.10)
- also updated some boto libs

### 🧪 Tested?

run crawlers locally and checked the metadata file

### CODE REVIEW

File: metaphor/common/extractor.py [hunk 1]
```diff
@@ -28,18 +29,30 @@ def config_class() -> Type[BaseConfig]:
     async def extract(self, config: BaseConfig) -> Collection[ENTITY_TYPES]:
         """Extract metadata and build messages, should be overridden"""
 
+    @abstractmethod
+    def platform(self) -> Optional[Platform]:
```
Comment (mars-lan) [inline metaphor/common/extractor.py:33]: Why optional?

Comment (alyiwang) [reply metaphor/common/extractor.py:33]: We didn't define platform for Slack and google directory, and manual lineage

File: pyproject.toml [hunk 2]
```diff
@@ -31,9 +31,9 @@ requests = "^2.27.1"
 slack-sdk = { version = "^3.5.1", optional = true }
 smart-open = "^5.0.0"
 snowflake-connector-python = { version = "~=2.7.0", optional = true }
-sql-metadata = { version = "2.3.0", optional = true }
+sql-metadata = { version = "^2.6.0", optional = true }
 tableauserverclient = { version = "^0.17.0", optional = true }
-thoughtspot-rest-api-sdk = { version = "1.11.0", optional = true }
+thoughtspot-rest-api-sdk = { version = "~1.11.0", optional = true }
```
Comment (elic-eon) [inline pyproject.toml:36]: 👍

File: metaphor/redshift/query/extractor.py [hunk 3]
```diff
@@ -20,11 +20,16 @@
 from metaphor.redshift.query.config import RedshiftQueryRunConfig
 
 logger = get_logger(__name__)
-logger.setLevel(logging.INFO)
 
 
 class RedshiftQueryExtractor(PostgreSQLExtractor):
-    """Redshift usage metadata extractor"""
+    """Redshift query extractor"""
```
Comment (elic-eon) [inline metaphor/redshift/query/extractor.py:26]: Thanks!

**Review by elic-eon — APPROVED** [review]
LGTM. Just curious that why some extractor import is not sorted?

File: metaphor/thought_spot/extractor.py [hunk 4]
```diff
@@ -53,6 +54,12 @@
 class ThoughtspotExtractor(BaseExtractor):
     """Thoughtspot metadata extractor"""
 
+    def platform(self) -> Optional[Platform]:
+        return Platform.THOUGHT_SPOT
+
+    def description(self) -> str:
+        return "Thought spot metadata crawler"
```
Comment (mars-lan) [inline metaphor/thought_spot/extractor.py:61]:
```suggestion
        return "ThoughtSpot metadata crawler"
```

**Comment by alyiwang** [comment]
> LGTM. Just curious that why some extractor import is not sorted?

I think by default `isort` only sorts imports within a group. Import groups are separated by blank lines, so if some imports are put in one group, they won't be automatically sorted to another group. The best practice is to have 3 groups: python native, third-party libraries, and other modules within the project.

**Comment by elic-eon** [comment]
> I think by default `isort` only sorts imports within a group. Import groups are separated by blank lines, so if some imports are put in one group, they won't be automatically sorted to another group. The best practice is to have 3 groups: python native, third-party libraries, and other modules within the project.

Cool, I thought `isort` will automatically group import into three by default but it didn't for some cases.

**Pull request merged**